extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,285 @@
+    msg.setSubject( "SUBJECT" );
+    msg.setText( "EmailService.MESSAGE" );
+    msg.setHeader( "X-Mailer", "smtpsend" );
+    msg.setSentDate( new Date() );
+    emailService.sendEmail( emailConfigOriginal, token, session, msg );
+  }
+
+  @Test( expected = HttpException.class )
+  public void testGetOAuthTokenAuthCode() throws HttpException, IOException {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "tes@test.com", "Pentaho Scheduler", "", 25,
+                    "graph_api", true, "test@test.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( "https://login.microsoftonline.com/common/oauth2/v2.0/token" );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( "authorization_code" );
+    emailConfigOriginal.setAuthorizationCode( "code" );
+    emailConfigOriginal.setRedirectUri( "uri" );
+    emailService.getOAuthToken( emailConfigOriginal );
+  }
+
+  @Test( expected = MessagingException.class )
+  public void testSendEmailSmtp() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "smtp.office365.com", 587,
+                    "SMTP", true, "PentahoBA_mailTest@Hitachivantara.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+
+    EmailAuthenticationResponse token = Mockito.mock( EmailAuthenticationResponse.class );
+    Properties emailProperties = new Properties();
+    emailProperties.setProperty( "mail.smtp.host", emailConfigOriginal.getSmtpHost() );
+    emailProperties.setProperty( "mail.smtp.port", ObjectUtils.toString( emailConfigOriginal.getSmtpPort() ) );
+    emailProperties.setProperty( "mail.transport.protocol", emailConfigOriginal.getSmtpProtocol() );
+    emailProperties.setProperty( "mail.smtp.starttls.enable", ObjectUtils.toString( emailConfigOriginal.isUseStartTls() ) );
+    emailProperties.setProperty( "mail.smtp.ssl", ObjectUtils.toString( emailConfigOriginal.isUseSsl() ) );
+    emailProperties.setProperty( "mail.debug", ObjectUtils.toString( emailConfigOriginal.isDebug() ) );
+    Session session = Session.getInstance( emailProperties );
+    MimeMessage msg = new MimeMessage( session );
+    msg.setFrom( new InternetAddress( emailConfigOriginal.getDefaultFrom(), emailConfigOriginal.getFromName() ) );
+    msg.setRecipients( Message.RecipientType.TO, InternetAddress.parse( emailConfigOriginal.getDefaultFrom() ) );
+    msg.setSubject( "SUBJECT" );
+    msg.setText( "EmailService.MESSAGE" );
+    msg.setHeader( "X-Mailer", "smtpsend" );
+    msg.setSentDate( new Date() );
+    emailService.sendEmail( emailConfigOriginal, token, session, msg );
```

Comment:
Should be using the MockMail approach to verify that the `msg` with the correct contents got sent to `Transport.send(...)`